protocols/relay/src/behaviour.rs

```diff
@@ -61,6 +61,32 @@ pub struct Config {
     pub circuit_src_rate_limiters: Vec<Box<dyn rate_limiter::RateLimiter>>,
 }
 
+impl Config {
+    pub fn per_peer(mut self, limit: NonZeroU32, interval: Duration) -> Self {
```

Comment:
Do we want these function to be per `reservation_rate_limiters` and `circuit_src_rate_limiters` or do we want to have it so we populate both vectors per function?

Reply:
I am not 100% sure I follow the question. I would expect a user to want to set a different limit for reservations than for circuits, with the rational that a reservation is less expensive than a circuit. Does that make sense? I am not sure though whether we really need to support this use-case via the easy-path, instead of deferring the user to the `rate-limiter` primitives directly.

Reply:
@mxinden see https://github.com/libp2p/rust-libp2p/pull/3742#issuecomment-1498853088.

I think splitting as better as per @mxinden's reasoning. We only provide these functions to be able to hide more details from the public API. The user should still be able to make fine-grained configuration changes otherwise there is no point in providing them I think.